AppCenter/AppCenter/Internals/Vendor/Reachability/MS_Reachability.m

```diff
@@ -124,7 +133,7 @@ + (instancetype)reachabilityForInternetConnection {
 #pragma mark - Start and stop notifier
 
 - (void)startNotifier {
-  dispatch_block_t block = ^{
+  RunOnMainThread(^{
     SCNetworkReachabilityContext context = {0, (__bridge void *)(self), NULL,
                                             NULL, NULL};
     if (SCNetworkReachabilitySetCallback(self.reachabilityRef,
```

Comment:
Also the pattern in using `self` in blocks is to use weak self as we do in other places in the SDK.

Reply:
Although if we use weakself the ref might never be unscheduled but will be freed so you are probably right not to use weakself. Just double check maybe.

Reply:
This approach lives since GA at least. It checked a lot of times